test/test_torch.py

```diff
@@ -1094,6 +1094,16 @@ def test_dtype_out_match(self):
         d = torch.autograd.Variable(torch.DoubleTensor(2, 3))
         self.assertRaises(RuntimeError, lambda: torch._C._VariableFunctions.zeros((2, 3), out=d, dtype=torch.float32))
 
+    def test_constructor_dtypes(self):
+        default_type = torch.Tensor().type()
+        self.assertIs(torch.Tensor().dtype, torch.Tensor.dtype)
+        torch.set_default_tensor_type('torch.IntTensor')
```

Comment:
Everything will end horribly if you use torch.IntTensor as the default type

Reply:
that's sort of why I chose it :) (I knew it wouldn't already be the default type).  This is just for a test and it's immediately set back to the original.